### DISCUSSION TITLE
How to run scripts using viem

### DISCUSSION THREAD

**Discussion by Confucian-PUT** [wevm/viem#469] [Question]
If I use hardhat, I could write scripts using ethers then simply run `npx hardhat run scripts/xx.ts` to interact with blockchian or smart-contract.
So is there a way to run viem as scripts? I try ts-node but failed.
I'm a new blockchain developer, thank you for your help!

**Comment by jxom** [comment]
Try `npx bun scripts/xx.ts`

**Reply by jxom** [reply]
(using [Bun](https://bun.sh))

**Reply by Confucian-PUT** [reply]
Well, I try to run `npx bun scripts/xx.ts` but there is an error:
```shell
1 | __require as require_ab102cbd93061951
2 | } from "bun:wrap";
3 | import * as $f1313928 from "/home/confucian/dev/viem-projects/attemp/node_modules/viem/dist/esm/errors/abi.js";
4 | 
5 | var {InvalidDefinitionTypeError} = require_ab102cbd93061951($f1313928);
   ^
ReferenceError: Cannot access uninitialized variable.
      at /home/confucian/dev/viem-projects/attemp/node_modules/viem/dist/esm/utils/abi/formatAbiItem.js:1:0
      at /home/confucian/dev/viem-projects/attemp/node_modules/viem/dist/esm/index.js:1:0
```
My ts file is: 
```typescript
import { createPublicClient, http } from "viem";
import { mainnet } from "viem/chains";

const client = createPublicClient({
    chain: mainnet,
    transport: http(),
});

const blockNumber = client.getBlockNumber();
console.log(blockNumber);
```

**Reply by Confucian-e** [reply]
> Well, I try to run `npx bun scripts/xx.ts` but there is an error:
> 
> ```shell
> 1 | __require as require_ab102cbd93061951
> 2 | } from "bun:wrap";
> 3 | import * as $f1313928 from "/home/confucian/dev/viem-projects/attemp/node_modules/viem/dist/esm/errors/abi.js";
> 4 | 
> 5 | var {InvalidDefinitionTypeError} = require_ab102cbd93061951($f1313928);
>    ^
> ReferenceError: Cannot access uninitialized variable.
>       at /home/confucian/dev/viem-projects/attemp/node_modules/viem/dist/esm/utils/abi/formatAbiItem.js:1:0
>       at /home/confucian/dev/viem-projects/attemp/node_modules/viem/dist/esm/index.js:1:0
> ```
> 
> My ts file is:
> 
> ```ts
> import { createPublicClient, http } from "viem";
> import { mainnet } from "viem/chains";
> 
> const client = createPublicClient({
>     chain: mainnet,
>     transport: http(),
> });
> 
> const blockNumber = client.getBlockNumber();
> console.log(blockNumber);
> ```

I meet the same error, how to fix this @jxom

**Comment by Confucian-PUT** [comment]
Thank you

**Comment by Confucian-PUT** [comment]
Resolved, I could run `npx ts-node scripts/xx.ts` to exceed, thank you for your time！
